test/common/config/metadata_test.cc

```diff
@@ -43,6 +48,77 @@ TEST(MetadataTest, MetadataValuePath) {
             ProtobufWkt::Value::KindCase::KIND_NOT_SET);
 }
 
+class TypedMetadataTest : public testing::Test {
+public:
+  TypedMetadataTest() : foo_factory_(), registered_factory_(foo_factory_) {}
+
+  struct Foo : public TypedMetadata::Object {
+    Foo(std::string name) : name_(name) {}
```

Comment:
I would make this `const std::string&`; this is just a nit, it doesn't matter for tests, but a good convention to stick to.